test/e2e/upgrade_management_cluster_test.go

```diff
@@ -251,6 +251,17 @@ func upgradeManagementCluster() {
 
 	By("THE MANAGEMENT CLUSTER WITH OLDER VERSION OF PROVIDERS WORKS!")
 
+	/*--------------------------------------*
+	| Upgrade Ironic and BareMetalOperator  |
+	*---------------------------------------*/
+
+	// TODO: If/when we rework the upgrade test to be able to use CAPI e2e
+	// upgrade test, these two should be included as a PreUpgrade step.
+	// If we do not go that route, we should instead refactor the whole
```

Comment:
I'm not sure what to call it, but I mean that we should break it up into smaller functions. For example, one to create the management cluster, one to pivot and one to do the capm3 upgrade. Then it would be easier to see what is going on and we would not have to call BMO and Ironic upgrades here inside.
But as I said in the comment, I think the best would be if we can use the CAPI test spec.